### ISSUE TITLE
help on the config options

### ISSUE THREAD

**Issue by imthenachoman** [dreamcat4/skippy-xd#11]
I know this project is not actively being maintained but since the application still works, is there anywhere I can find an explanation of all of the config options? I am playing around with some but can't find what they mean.

**Comment by dreamcat4** [comment]
I would like to help but have other pressing matters to deal with. I believe that not all options are equally useful as each other but if you are having trouble understanding a specific option then it would be best to say which one(s) and be more specific about it.

Also not all options have a meaningful effect in all modes. Some combinations of options will therefore hide the behaviour of the weaker option.

**Comment by imthenachoman** [comment]
I completely understand. Don't stress yourself over this. I figure maybe others might chime in if they have time. I've got it installed and it works great. I just couldn't figure out a few things.

For example, in the comments, it has `#FF000080` for `background`. I assume that `80` is for the alpha channel (transparency). However, it does not work for me. I'm not sure if I'm doing something wrong. I'm trying to get a black background with 50% transparency.

I've been playing around with the other settings. I'll put a table together of all the options and what I understand them to be for. It wont be complete but maybe, as time goes on, other users can fill in with what they learn. 

Again, I completely understand being busy and its okay if you don't respond. I totally get it.

**Comment by dreamcat4** [comment]
Hey there. I think I see your problem now...  Thank you for explaining this. Well looking in the source code at the place where the config setting is parsed and loaded into the program.... it says here:

https://github.com/richardgv/skippy-xd/blob/master/src/skippy.c#L1259-L1282

That it's a `pictspec` and also beneath that is the same for the `iconFillSpec` which has the multi word format something like `orig mid mid #FFFFFF`.

And here we can go back to the config file here (in master branch) and see it shows an example of it already:

https://github.com/dreamcat4/skippy-xd/blob/master/skippy-xd.sample.rc#L26-L32

And the example has an either / or parsing option those 2 extra numbers in the color. Or otherwise a normal color.

We can see where this is parsed in the code here:

https://github.com/richardgv/skippy-xd/blob/master/src/skippy.c#L249

Goes to:

https://github.com/richardgv/skippy-xd/blob/master/src/skippy.c#L112-L168

Which... as you appear to be correct that the last byte `80` at the end is being masked as a 256 levels alpha channel designated as `AA`.

https://github.com/richardgv/skippy-xd/blob/master/src/skippy.c#L160

Otherwise the alpha is being set by default to `0xffff`

So as an experiment (what I would do, but dont have the time here) is change that hardcoded value in the src code. And recompile a test version of skippy executable. And see if that changed the alpha. Then you know the if statement is not working right. And it's defaulting there always. Even if it should not.

You are correct to raise this issue. As it would seem to be a genuine bug in the code. That is not working as expected.

The above simple test will then ascertain if the bug is right there (in that place). Or if the alpha setting variable is loaded and set correctly. In which case the broken alpha (not working) is instead occuring in some other place in the skippy's code. For example at the time when the background is being created / being drawn.

This is also (like the other recent issue). A very good introduction to the skippy source code. AS it's a small problem that almost anybody can try to debug. And attempt to fix. And during that process  also begin building a good foundation knowledge about how skippy is working under the hood. It is really a very traditional 'old school' X windows application / C program. And whilst a bit messy with many such bugs. Can also be very rewarding to work on. Especially if you want to takle some real world C code. That is a workhorse and see directly on the screen the results of your efforts. It is very rewarding process.

I encourage anybody interested in skippy to try to fix this bug. And I am also here from time to time if I can provide any support. From what I know. Which is really not everything about skippy!

Most of the time it's tracing through the source code. To follow what the path of execution is. With `ctrl+f` and having a 'vague idea' which parts are generally supposed to do which steps / stages.

**Comment by dreamcat4** [comment]
Another help for debugging is simply to insert a `printf()` trace line statement manually (or whatever the debug logline macro is in skippy). To just print out at that place the alpha value of the setting.

And also maybe once more in the totally different place in skippy which is making the graphic call to be creating the background window. And check the value against what you are specifying in your config file.

And be sure that skippy is picking up the right config file that you want it too. (Sometimes it might not be reading from that expected location, depending on the system priorities where it looks, or if the config file is installed to the correct location).

OK Thanks again for you understanding. Hope you can make a progress.

**Comment by imthenachoman** [comment]
This is great information. I don't know C (mostly JS development) but I'll see if I can step through this. First I'll have to figure out how to compile from source. Then I can muck around with settings and see what happens. Thanks!

**Comment by dreamcat4** [comment]
Yeah you shouldn't have much trouble compiling. As skippy doesnt have many external dependencies. And a text editor.

**Comment by ozls** [comment]
I got here because I was facing the same bug. Did you manage to solve this issue ? If yes, I would really appreciate your fix.
